frontend/src/scenes/ContactsPage.tsx

```diff
@@ -438,46 +444,71 @@ function EditContactDialog({
   );
 }
 
+function ContactsCountInfo({
+  contacts,
+  locale,
+}: {
+  contacts: any[];
+  locale: Locale;
+}) {
+  return (
+    <div className="info-container" style={{ marginTop: '12px' }}>
+      <InfoItem variant="subtitle2" className="info-item">
+        <FavoriteIcon className="info-icon" />
+        {contacts.length} {STRINGS[locale].CONTACTS_LOVED_ONES_LABEL}
+      </InfoItem>
+    </div>
+  );
+}
+
 function CallLimitInfo({
   userCallTimeDuration,
-  contacts,
   locale,
   nextRefreshTimeString,
   nextRefreshAmount,
   callLimitExceeded,
 }: {
   userCallTimeDuration: Duration;
-  contacts: any[];
   locale: Locale;
   nextRefreshTimeString: string | null;
   nextRefreshAmount: string | null;
   callLimitExceeded: boolean;
 }) {
-  const CallTimeInfoItem = callLimitExceeded ? ErrorInfoItem : InfoItem;
+  const CallTimeInfoItem = callLimitExceeded ? ErrorInfoItem : LightInfoItem;
+  const callTimeInMins: number = userCallTimeDuration.as('minutes');
+  const callTimeRatio: number = Math.min(
+    Math.max(callTimeInMins / CALL_TIME_TOTAL_MINS, 0),
+    1
+  );
+
   // TODO localization problems here
-  const nextRefreshInfoElement =
-    nextRefreshTimeString && nextRefreshAmount ? (
-      <p>
-        <strong>{nextRefreshAmount}</strong> in {nextRefreshTimeString}
-      </p>
-    ) : (
-      ''
-    );
+
   return (
-    <div className="info-container" style={{ marginTop: '12px' }}>
-      <CallTimeInfoItem variant="subtitle2" className="info-item">
-        <PhoneInTalkIcon className="info-icon" />
-        {formatDurationInHoursMinutes(userCallTimeDuration)}
-      </CallTimeInfoItem>
+    <div style={{ marginTop: '12px' }}>
+      <RoundedProgressBar
+        variant="determinate"
+        value={callTimeRatio * 100}
+        style={{ width: '100%' }}
+      />
 
-      <InfoItem variant="subtitle2" className="info-item">
-        <AutorenewIcon className="info-icon" />
-        {nextRefreshInfoElement}
-      </InfoItem>
-      <InfoItem variant="subtitle2" className="info-item">
-        <FavoriteIcon className="info-icon" />
-        {contacts.length} {STRINGS[locale].CONTACTS_LOVED_ONES_LABEL}
-      </InfoItem>
+      <div
+        className="info-container"
+        style={{ justifyContent: 'space-between', marginTop: 4 }}
+      >
+        <CallTimeInfoItem variant="body2">
+          {formatDurationInMinutes(userCallTimeDuration)} left
+        </CallTimeInfoItem>
+
+        {nextRefreshTimeString && nextRefreshAmount && (
+          <LightInfoItem
+            variant="body2"
+            style={{ display: 'flex', alignItems: 'center' }}
+          >
+            <RefreshIcon style={{ marginRight: 4 }} />+{nextRefreshAmount} in{' '}
```

Comment:
Try to avoid putting literal strings in the template directly. It will affect localisation.